File: md-ml-py/matmul_party_1.py
"""Matrix multiplication experiment - party 1."""

from md_ml.share import Spdz2kShare64
from md_ml.protocols import PartyWithFakeOffline, Circuit

FAKE_OFFLINE_DIR = "../fake-offline-data"
JOB_NAME = "MatMul-py"
DIM = 4096
PORT = 7070


def main():
    share = Spdz2kShare64

    party = PartyWithFakeOffline(share, 1, 2, PORT, JOB_NAME, FAKE_OFFLINE_DIR)
    circuit = Circuit(share)

    a = circuit.input(0, DIM, DIM)
    b = circuit.input(0, DIM, DIM)
    c = circuit.multiply(a, b)
    d = circuit.output(c)
    circuit.add_endpoint(d)

    circuit.read_offline_from_file(party)
    circuit.run_online_with_benchmark(party)
    circuit.print_stats(party)


if __name__ == "__main__":
    main()

File: md-ml-rs/src/bin/matmul_party_1.rs
// Matrix multiplication experiment - party 1 (Rust rewrite)

use md_ml::share::Spdz2kShare64;
use md_ml::protocols::{PartyWithFakeOffline, Circuit};

const FAKE_OFFLINE_DIR: &str = "../fake-offline-data";
const JOB_NAME: &str = "MatMul";
const DIM: usize = 4096;

fn main() {
    type S = Spdz2kShare64;

    let mut party = PartyWithFakeOffline::<S>::new(1, 2, 6060, JOB_NAME, FAKE_OFFLINE_DIR);
    let mut circuit = Circuit::<S>::new();

    let a = circuit.input(0, DIM, DIM);
    let b = circuit.input(0, DIM, DIM);
    let c = circuit.multiply(a, b);
    let d = circuit.output(c);
    circuit.add_endpoint(d);

    circuit.read_offline_from_file(&mut party);
    circuit.run_online_with_benchmark(&mut party);
    circuit.print_stats(&party);
}
